app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,53 @@
+<% content_for :head do %>
+  <%= javascript_pack_tag 'togglePublic', 'data-turbolinks-track': 'reload' %>
+<% end %>
+
+<% if can? :read, @recipe %>
+  <div class='recipe-detail-container'>
+    <div class='recipe-detail-header'>
+      <h2 class='recipe-detail-name'>
+        <%= @recipe.name %>
+      </h2>
+
+      <div class="recipe-detail-prep">
+        <p>
+          Preparation time: <%= @recipe.preparation_time %> hour
+        </p>
+        <% if user_signed_in? %>
+          <div class='recipe-detail-public'>
+            <p>Public</p>
+            <%= render partial: '_partials/forms/toggle_public' %>
+          </div>
+        <% else %>
+          <label class="switch">
+            <input type="checkbox" checked/>
+            <span class="slider round"></span>
+          </label>
+        <% end %>
+      </div>
+      <p>
+        Cooking time: <%= @recipe.cooking_time %> hour
+      </p>
+      <p>
+        <%= @recipe.description %>
+      <p>
+    </div>
+    <div class='recipe-detail-buttons'>
+      <% if user_signed_in? %>
+        <%= link_to general_shopping_lists_path  do %>
+          <button type="button" class='detail-button'>Generate shopping list</button>
```

Comment:
- [ ] On the recipe add ingredient page, the dropdown select field for food is empty whereas I have foods available, I consider this as a bug, would you fix it?
![Screenshot from 2022-01-08 17-09-57](https://user-images.githubusercontent.com/56886064/148649763-e2f45b96-b48e-4972-9fa1-ec6b1560afc9.png)

- [ ] When clicking on the generate shopping list, the following interface should appear
![Screenshot from 2022-01-08 17-12-02](https://user-images.githubusercontent.com/56886064/148649773-abe935e7-fa2d-445b-9893-31f72277c6c9.png)

Reply:
@descholar-ceo Thank you for your review, there is one detail about this though. We are only a group of 2 persons so we are not using inventories. Please let us know how to proceed, thank you!